recipes/upgradetool_run.rb

```diff
@@ -0,0 +1,27 @@
+#
+# Cookbook Name:: cdap
+# Recipe:: upgradetool_run
+#
+# Copyright © 2015 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include_recipe 'cdap::master'
```

Comment:
You should save the configured `init_actions` for master, override it to stop, include_recipe master, run tool, then notify the master service with the initial `init_actions` for master.